diff -ruBw xproto-7.0.26-a/X.h xproto-7.0.26/X.h
--- xproto-7.0.26-a/X.h	2013-06-09 18:01:47.000000000 +0800
+++ xproto-7.0.26/X.h	2014-05-17 17:52:26.703400000 +0800
@@ -63,18 +63,18 @@
 #ifndef _XSERVER64
 #  ifndef _XTYPEDEF_XID
 #    define _XTYPEDEF_XID
-typedef unsigned long XID;
+typedef unsigned long long XID;
 #  endif
 #  ifndef _XTYPEDEF_MASK
 #    define _XTYPEDEF_MASK
-typedef unsigned long Mask;
+#        define Mask unsigned long long
 #  endif
 #  ifndef _XTYPEDEF_ATOM
 #    define _XTYPEDEF_ATOM
-typedef unsigned long Atom;		/* Also in Xdefs.h */
+typedef unsigned long long Atom;		/* Also in Xdefs.h */
 #  endif
-typedef unsigned long VisualID;
-typedef unsigned long Time;
+typedef unsigned long long VisualID;
+typedef unsigned long long Time;
 #else
 #  include <X11/Xmd.h>
 #  ifndef _XTYPEDEF_XID
diff -ruBw xproto-7.0.26-a/Xwindows.h xproto-7.0.26/Xwindows.h
--- xproto-7.0.26-a/Xwindows.h	2014-03-27 06:32:15.000000000 +0800
+++ xproto-7.0.26/Xwindows.h	2014-05-15 19:24:17.129800000 +0800
@@ -65,14 +65,25 @@
 #define INT32 wINT32
 #ifdef __x86_64__
 #define INT64 wINT64
+#ifdef LONG64
+#define tLONG64 LONG64
+#undef LONG64
+#endif
 #define LONG64 wLONG64
 #endif
 #undef Status
 #define Status wStatus
+#ifdef Mask
+#define tMask Mask
+#undef Mask
+#endif
 #define ATOM wATOM
 #define BYTE wBYTE
 #define FreeResource wFreeResource
 #include <windows.h>
+#ifdef tMask
+#define Mask tMask
+#endif
 #undef NOMINMAX
 #undef Status
 #define Status int
@@ -81,6 +92,9 @@
 #undef INT32
 #undef INT64
 #undef LONG64
+#ifdef tLONG64
+#define LONG64 tLONG64
+#endif
 #undef ATOM
 #undef FreeResource
 #undef CreateWindowA
diff -ruBw xproto-7.0.26-a/Xwinsock.h xproto-7.0.26/Xwinsock.h
--- xproto-7.0.26-a/Xwinsock.h	2013-06-09 18:08:14.000000000 +0800
+++ xproto-7.0.26/Xwinsock.h	2014-05-15 19:36:24.483600000 +0800
@@ -47,17 +47,38 @@
 #define _NO_BOOL_TYPEDEF
 #define BOOL WINBOOL
 #define INT32 wINT32
+#ifdef __x86_64__
+#define INT64 wINT64
+#ifdef LONG64
+#define tLONG64 LONG64
+#undef LONG64
+#endif
+#define LONG64 wLONG64
+#endif
 #undef Status
+#ifdef Mask
+#define tMask Mask
+#undef Mask
+#endif
 #define Status wStatus
 #define ATOM wATOM
 #define BYTE wBYTE
 #define FreeResource wFreeResource
 #include <winsock2.h>
+#include <ws2tcpip.h>
+#ifdef tMask
+#define Mask tMask
+#endif
 #undef Status
 #define Status int
 #undef BYTE
 #undef BOOL
 #undef INT32
+#undef INT64
+#undef LONG64
+#ifdef tLONG64
+#define LONG64 tLONG64
+#endif
 #undef ATOM
 #undef FreeResource
 #undef CreateWindowA
