diff -ruBw libXft-2.3.1-a/src/xftglyphs.c libXft-2.3.1/src/xftglyphs.c
--- libXft-2.3.1-a/src/xftglyphs.c	2012-06-03 00:36:35.000000000 +0800
+++ libXft-2.3.1/src/xftglyphs.c	2014-05-15 21:05:11.453800000 +0800
@@ -21,10 +21,10 @@
  */
 
 #include "xftint.h"
-#include <freetype/ftoutln.h>
-#include <freetype/ftlcdfil.h>
-
-#include <freetype/ftsynth.h>
+#include <ft2build.h>
+#include FT_OUTLINE_H
+#include FT_LCD_FILTER_H
+#include FT_SYNTHESIS_H
 
 /*
  * Validate the memory info for a font
@@ -927,7 +927,7 @@
 	{
 	    if (font->glyph_memory > font->max_glyph_memory)
 		printf ("Reduce memory for font 0x%lx from %ld to %ld\n",
-			font->glyphset ? font->glyphset : (unsigned long) font,
+			font->glyphset ? font->glyphset : (unsigned long) ((intptr_t) font),
 			font->glyph_memory, font->max_glyph_memory);
 	}
 	while (font->glyph_memory > font->max_glyph_memory)
