diff --git a/include/X11/Xlib.h b/include/X11/Xlib.h
index 2bffa76..3be3af0 100644
--- a/include/X11/Xlib.h
+++ b/include/X11/Xlib.h
@@ -1385,7 +1385,7 @@ typedef struct {
 
 _XFUNCPROTOBEGIN
 
-#if defined(WIN32) && !defined(_XLIBINT_)
+#if defined(WIN32) && !defined(__MINGW32__) && !defined(_XLIBINT_)
 #define _Xdebug (*_Xdebug_p)
 #endif
 
diff --git a/include/X11/Xlibint.h b/include/X11/Xlibint.h
index 4431559..d5f114d 100644
--- a/include/X11/Xlibint.h
+++ b/include/X11/Xlibint.h
@@ -273,7 +273,7 @@ struct _XLockPtrs {
 	);
 };
 
-#if defined(WIN32) && !defined(_XLIBINT_)
+#if defined(WIN32) && !defined(__MINGW32__) && !defined(_XLIBINT_)
 #define _XCreateMutex_fn (*_XCreateMutex_fn_p)
 #define _XFreeMutex_fn (*_XFreeMutex_fn_p)
 #define _XLockMutex_fn (*_XLockMutex_fn_p)
diff --git a/src/XlibInt.c b/src/XlibInt.c
index 80c1298..635e3ba 100644
--- a/src/XlibInt.c
+++ b/src/XlibInt.c
@@ -31,7 +31,7 @@ from The Open Group.
  *	interface library (Xlib) to the X Window System Protocol V11.0.
  */
 
-#ifdef WIN32
+#if defined(WIN32) && !defined(__MINGW32__)
 #define _XLIBINT_
 #endif
 #ifdef HAVE_CONFIG_H
@@ -1735,11 +1735,7 @@ void *_XGetRequest(Display *dpy, CARD8 type, size_t len)
  * be converted to "C:/usr/X11R6.3/lib/X11/locale/locale.dir".
  */
 
-static int access_file (path, pathbuf, len_pathbuf, pathret)
-    char* path;
-    char* pathbuf;
-    int len_pathbuf;
-    char** pathret;
+static int access_file (const char *path, char *pathbuf, int len_pathbuf, char **pathret)
 {
     if (access (path, F_OK) == 0) {
 	if (strlen (path) < len_pathbuf)
@@ -1754,15 +1750,11 @@ static int access_file (path, pathbuf, len_pathbuf, pathret)
     return 0;
 }
 
-static int AccessFile (path, pathbuf, len_pathbuf, pathret)
-    char* path;
-    char* pathbuf;
-    int len_pathbuf;
-    char** pathret;
+static int AccessFile (const char *path, char *pathbuf, int len_pathbuf, char **pathret)
 {
     unsigned long drives;
     int i, len;
-    char* drive;
+    const char* drive;
     char buf[MAX_PATH];
     char* bufp;
 
@@ -1827,15 +1819,14 @@ static int AccessFile (path, pathbuf, len_pathbuf, pathret)
     return 0;
 }
 
-int _XOpenFile(path, flags)
-    _Xconst char* path;
-    int flags;
+int _XOpenFile(_Xconst char *path, int flags)
 {
     char buf[MAX_PATH];
     char* bufp = NULL;
     int ret = -1;
     UINT olderror = SetErrorMode (SEM_FAILCRITICALERRORS);
 
+
     if (AccessFile (path, buf, MAX_PATH, &bufp))
 	ret = open (bufp, flags);
 
@@ -1846,10 +1837,7 @@ int _XOpenFile(path, flags)
     return ret;
 }
 
-int _XOpenFileMode(path, flags, mode)
-    _Xconst char* path;
-    int flags;
-    mode_t mode;
+int _XOpenFileMode(_Xconst char *path, int flags, mode_t mode)
 {
     char buf[MAX_PATH];
     char* bufp = NULL;
@@ -1866,9 +1854,7 @@ int _XOpenFileMode(path, flags, mode)
     return ret;
 }
 
-void* _XFopenFile(path, mode)
-    _Xconst char* path;
-    _Xconst char* mode;
+void* _XFopenFile(_Xconst char *path, _Xconst char *mode)
 {
     char buf[MAX_PATH];
     char* bufp = NULL;
@@ -1885,8 +1871,7 @@ void* _XFopenFile(path, mode)
     return ret;
 }
 
-int _XAccessFile(path)
-    _Xconst char* path;
+int _XAccessFile(_Xconst char *path)
 {
     char buf[MAX_PATH];
     char* bufp;
@@ -1904,7 +1889,7 @@ int _XAccessFile(path)
 
 #endif
 
-#ifdef WIN32
+#if defined(WIN32) && !defined(__MINGW32__)
 #undef _Xdebug
 int _Xdebug = 0;
 int *_Xdebug_p = &_Xdebug;
diff --git a/src/locking.c b/src/locking.c
index 9f4fe06..448b699 100644
--- a/src/locking.c
+++ b/src/locking.c
@@ -67,7 +67,7 @@ in this Software without prior written authorization from The Open Group.
 /* in lcWrap.c */
 extern LockInfoPtr _Xi18n_lock;
 
-#ifdef WIN32
+#if defined(WIN32) && !defined(__MINGW32__)
 static DWORD _X_TlsIndex = (DWORD)-1;
 
 void _Xthread_init(void)
