diff --git a/X.h b/X.h
index 5cf695d..a1199a2 100644
--- a/X.h
+++ b/X.h
@@ -63,18 +63,31 @@ SOFTWARE.
 #ifndef _XSERVER64
 #  ifndef _XTYPEDEF_XID
 #    define _XTYPEDEF_XID
+#    ifdef __x86_64__
+typedef unsigned long long XID;
+#    else
 typedef unsigned long XID;
+#    endif
 #  endif
 #  ifndef _XTYPEDEF_MASK
 #    define _XTYPEDEF_MASK
-typedef unsigned long Mask;
+#      define Mask unsigned long
 #  endif
 #  ifndef _XTYPEDEF_ATOM
 #    define _XTYPEDEF_ATOM
-typedef unsigned long Atom;		/* Also in Xdefs.h */
+#      ifdef __x86_64__
+typedef unsigned long long Atom;		/* Also in Xdefs.h */
+#      else
+typedef unsigned long Atom;
+#      endif
 #  endif
+#  ifdef __x86_64__
+typedef unsigned long long VisualID;
+typedef unsigned long long Time;
+#  else
 typedef unsigned long VisualID;
 typedef unsigned long Time;
+#  endif
 #else
 #  include <X11/Xmd.h>
 #  ifndef _XTYPEDEF_XID
diff --git a/Xthreads.h b/Xthreads.h
index 2027127..355ca5f 100644
--- a/Xthreads.h
+++ b/Xthreads.h
@@ -94,7 +94,7 @@ extern xthread_t (*_x11_thr_self)();
 #   define xcondition_signal(cv) cond_signal(cv)
 #   define xcondition_broadcast(cv) cond_broadcast(cv)
 #  else /* !SVR4 */
-#   ifdef WIN32
+#   if defined(WIN32) && !defined(__MINGW32__)
 #    include <X11/Xwindows.h>
 typedef DWORD xthread_t;
 typedef DWORD xthread_key_t;
@@ -260,6 +260,9 @@ static xthread_t _X_no_thread_id;
 #       define xthread_clear_id(id) id = _X_no_thread_id
 #       define xthread_equal(id1,id2) pthread_equal(id1, id2)
 #      endif /* _DECTHREADS_ */
+#      if defined(__MINGW32__)
+#       define xthread_equal(id1,id2) pthread_equal(id1, id2)
+#      endif /* MINGW32 */
 #      if defined(__linux__)
 #       define xthread_have_id(id) !pthread_equal(id, 0)
 #       define xthread_clear_id(id) id = 0
diff --git a/Xwindows.h b/Xwindows.h
index 70e1deb..5f8eb93 100644
--- a/Xwindows.h
+++ b/Xwindows.h
@@ -64,16 +64,21 @@ The Open Group.
 #define BOOL WINBOOL
 #define INT32 wINT32
 #ifdef __x86_64__
-#define INT64 wINT64
-#define LONG64 wLONG64
-#endif
+#  define INT64 wINT64
+#  pragma push_macro("LONG64")
+#  undef LONG64
+#  define LONG64 wLONG64
+#endif /* __x86_64__ */
 #undef Status
 #define Status wStatus
+#pragma push_macro("Mask")
+#undef Mask
 #define ATOM wATOM
 #define BYTE wBYTE
 #define FreeResource wFreeResource
 #include <windows.h>
 #undef NOMINMAX
+#pragma pop_macro("Mask")
 #undef Status
 #define Status int
 #undef BYTE
@@ -81,6 +86,7 @@ The Open Group.
 #undef INT32
 #undef INT64
 #undef LONG64
+#pragma pop_macro("LONG64")
 #undef ATOM
 #undef FreeResource
 #undef CreateWindowA
diff --git a/Xwinsock.h b/Xwinsock.h
index a81dd7a..d710f8b 100644
--- a/Xwinsock.h
+++ b/Xwinsock.h
@@ -47,17 +47,29 @@ The Open Group.
 #define _NO_BOOL_TYPEDEF
 #define BOOL WINBOOL
 #define INT32 wINT32
+#ifdef __x86_64__
+#  define INT64 wINT64
+#  pragma push_macro("LONG64")
+#  undef LONG64
+#  define LONG64 wLONG64
+#endif /* __x86_64__ */
 #undef Status
 #define Status wStatus
+#pragma push_macro("Mask")
+#undef Mask
 #define ATOM wATOM
 #define BYTE wBYTE
 #define FreeResource wFreeResource
 #include <winsock2.h>
+#pragma pop_macro("Mask")
 #undef Status
 #define Status int
 #undef BYTE
 #undef BOOL
 #undef INT32
+#undef INT64
+#undef LONG64
+#pragma pop_macro("LONG64")
 #undef ATOM
 #undef FreeResource
 #undef CreateWindowA
