diff -ruBw libXext-1.3.2-a/src/XAppgroup.c libXext-1.3.2/src/XAppgroup.c
--- libXext-1.3.2-a/src/XAppgroup.c	2013-06-01 05:35:08.000000000 +0800
+++ libXext-1.3.2/src/XAppgroup.c	2014-05-15 20:48:06.999400000 +0800
@@ -337,7 +337,7 @@
 XagCreateAssociation(Display* dpy, Window* window_return, void* system_window)
 {
 #ifdef WIN32
-    long tmp = *(HWND*) system_window;
+    HWND tmp = *(HWND*) system_window;
     XExtDisplayInfo *info = find_display (dpy);
     xXagCreateAssocReq *req;
 
@@ -350,7 +350,7 @@
     *window_return = req->window = XAllocID(dpy);
     req->window_type = XagWindowTypeWin32;
     req->system_window_len = sizeof(HWND);
-    Data32 (dpy, (long*) tmp, 1L);
+    Data32 (dpy, tmp, 1L);
     req->length++;
     UnlockDisplay(dpy);
     SyncHandle();
